### PR TITLE
pybind/mgr/devicehealth: skip legacy objects that cannot be loaded

### PR DESCRIPTION
**Pull request by batrick**
Fixes: https://tracker.ceph.com/issues/63882

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by batrick** [comment]
jenkins test api

**Comment by batrick** [comment]
(trivial rebase)

**Comment by batrick** [comment]
jenkins test api

File: qa/tasks/mgr/mgr_test_case.py [hunk 1]
```diff
@@ -89,6 +92,9 @@ def setup_mgrs(cls):
             cls.mgr_cluster.mon_manager.raw_cluster_cmd(
                 "mgr", "module", "disable", m)
 
+        cls.mgr_cluster.mon_manager.raw_cluster_cmd('config', 'set', 'mon', 'mon_allow_pool_delete', 'true')
+        cls.mgr_cluster.mon_manager.raw_cluster_cmd('osd', 'pool', 'rm', '.mgr', '.mgr', '--yes-i-really-really-mean-it-not-faking')
```
Comment (yaarith) [inline qa/tasks/mgr/mgr_test_case.py:96]: I was wondering if there are any scenarios where the .mgr pool needs to persist between tests; I guess not, but just a thought

**Review by yaarith — APPROVED** [review]
Looks good

**Comment by yaarith** [comment]
jenkins retest this please

**Comment by batrick** [comment]
jenkins test api

File: qa/tasks/mgr/mgr_test_case.py (hunk 1 above)
Comment (batrick) [reply qa/tasks/mgr/mgr_test_case.py:96]: It's a valid concern but I think in this case it's appropriate for testing the mgr modules.

**Comment by batrick** [comment]
jenkins test api

**Comment by ljflores** [comment]
jenkins test api

**Comment by ljflores** [comment]
jenkins test api

**Comment by ljflores** [comment]
Rados-approved: https://tracker.ceph.com/projects/rados/wiki/MAIN#httpstrellocomchaVLXC6K1922-wip-yuri10-testing-2024-01-04-1245-old-wip-yuri10-testing-2024-01-03-1546

**Comment by yuriw** [comment]
jenkins test api

**Comment by yuriw** [comment]
@batrick pls merge when api tests passed

ref: https://trello.com/c/haVLXC6K

**Comment by batrick** [comment]
> @batrick pls merge when api tests passed
> 
> ref: https://trello.com/c/haVLXC6K

It looks like the dashboard test may have  failed due to these changes. I'll see if I can reproduce.

**Comment by batrick** [comment]
jenkins test api

**Comment by batrick** [comment]
@yaarith please have another review. I made several changes to avoid deleting the `.mgr` pool so that the dashboard tests do not break.

@yuriw  / @ljflores this needs a new round of QA.

**Comment by yaarith** [comment]
The fix looks good to me.
@kamoltat @NitzanMordhai can you please review? There are several mon changes.
Please note that this fix should be backported to reef and quincy.

**Review by NitzanMordhai — APPROVED** [review]
LGTM

**Pull request merged**